tests/test_benchmark_simple_scenarios.py

```diff
@@ -190,32 +196,56 @@ def test_benchmark_AE_grid_battery_peak_pricing(self, margs):
             os.path.join(TEST_OUTPUT_PATH, use_case, "json_with_results.json"), "r"
         ) as results:
             data = json.load(results)
-        peak_demand = data["energyConversion"][
-            "Electricity grid DSO_consumption_period_1"
-        ]["optimizedAddCap"]["value"]
+        peak_demand = [
+            data[ENERGY_CONVERSION]["Electricity grid DSO_consumption_period_1"][
+                OPTIMIZED_ADD_CAP
+            ][VALUE],
+            data[ENERGY_CONVERSION]["Electricity grid DSO_consumption_period_2"][
+                OPTIMIZED_ADD_CAP
+            ][VALUE],
+            data[ENERGY_CONVERSION]["Electricity grid DSO_consumption_period_2"][
+                OPTIMIZED_ADD_CAP
+            ][VALUE],
+        ]
         # read timeseries_all_busses excel file
         busses_flow = pd.read_excel(
             os.path.join(TEST_OUTPUT_PATH, use_case, "timeseries_all_busses.xlsx"),
             sheet_name=bus_suffix("Electricity"),
         )
         # make the time the index
         busses_flow = busses_flow.set_index("Unnamed: 0")
-        # save the columns with the values to be used
-        DSO_period_1 = busses_flow["Electricity grid DSO_consumption_period_1"]
+        # read the columns with the values to be used
+        DSO_period = [
+            busses_flow["Electricity grid DSO_consumption_period_1"],
+            busses_flow["Electricity grid DSO_consumption_period_2"],
+            busses_flow["Electricity grid DSO_consumption_period_3"],
+        ]
         demand = busses_flow["demand_01"]
         battery_charge = busses_flow["battery"]
         battery_discharge = (
             abs(busses_flow["demand_01"])
             - busses_flow["Electricity grid DSO_consumption_period_1"]
-        )
-        # look for peak demand in column
-        for i in range(0, len(DSO_period_1)):
-            if DSO_period_1[i] == peak_demand and abs(demand[i]) < DSO_period_1[i]:
-                assert battery_charge[i] < 0
-            if DSO_period_1[i] == peak_demand and abs(demand[i]) > DSO_period_1[i]:
-                assert battery_discharge[i] > 0
-            if DSO_period_1[i] == peak_demand and DSO_period_1[i - 1] != peak_demand:
-                assert battery_charge[i - 1] < 0
+        )  # todo: replace this by discharge column when implemented
+        # pick few instances
+        instances = [random.randint(1, len(DSO_period[1])) for x in range(0, 60)]
```

Comment:
Hm, this would choose any date in the whole DSO_Period_1 timeseries - and 60% of these are of value 0. I think I would like to go back to the way you did it before after all. It also has the added benefit, that we can for sure say that this is true for EVERY TIMESTEP. Better then saying for a random set of x timestep, this is true.